v1/src/test/java/com/google/cloud/teleport/bigtable/AvroToBigtablePerformanceIT.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (C) 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.bigtable;
+
+import static com.google.cloud.teleport.it.artifacts.ArtifactUtils.createGcsClient;
+import static com.google.cloud.teleport.it.artifacts.ArtifactUtils.getFullGcsPath;
+import static com.google.cloud.teleport.it.dataflow.DataflowUtils.createJobName;
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.storage.Storage;
+import com.google.cloud.teleport.it.DataGenerator;
+import com.google.cloud.teleport.it.PerformanceBenchmarkingBase;
+import com.google.cloud.teleport.it.TestProperties;
+import com.google.cloud.teleport.it.artifacts.ArtifactClient;
+import com.google.cloud.teleport.it.artifacts.GcsArtifactClient;
+import com.google.cloud.teleport.it.bigtable.BigtableResourceManager;
+import com.google.cloud.teleport.it.bigtable.DefaultBigtableResourceManager;
+import com.google.cloud.teleport.it.dataflow.DataflowClient.JobInfo;
+import com.google.cloud.teleport.it.dataflow.DataflowClient.JobState;
+import com.google.cloud.teleport.it.dataflow.DataflowClient.LaunchConfig;
+import com.google.cloud.teleport.it.dataflow.DataflowOperator.Result;
+import com.google.cloud.teleport.metadata.TemplateIntegrationTest;
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.Resources;
+import java.io.IOException;
+import java.text.ParseException;
+import java.time.Duration;
+import java.util.Map;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Performance tests for {@link AvroToBigtable} (GCS_Avro_to_Cloud_Bigtable) template. */
+@TemplateIntegrationTest(AvroToBigtable.class)
+@RunWith(JUnit4.class)
+public class AvroToBigtablePerformanceIT extends PerformanceBenchmarkingBase {
+  private static final Logger LOG = LoggerFactory.getLogger(AvroToBigtablePerformanceIT.class);
+  private static final String SPEC_PATH =
+      MoreObjects.firstNonNull(
+          "gs://dataflow-templates/latest/GCS_Avro_to_Cloud_Bigtable", TestProperties.specPath());
```

Comment:
something to think / longer term - do you think we can try to mimic TemplateTestBase to stage those templates on demand? Or make that an option/property? It would be good to validate benchmark across changes of a template, instead of using a fixed baseline (both are good, but we can optionally set `-DspecPath=gs://dataflow-templates/latest/GCS_Avro_to_Cloud_Bigtable` like handled here)

Reply:
I'm not sure we should do that since we are planning on moving the IT module to the beam repo soon? PerformanceBenchmarkingBase will be used for non-templates tests too